### ISSUE TITLE
Make tide the only required check for PRs

### ISSUE THREAD

**Issue by chaodaiG** [knative/test-infra#1742]
As part of knative repo setup steps, it's required to [check all required Prow jobs](https://github.com/knative/test-infra/blob/master/guides/prow_setup.md#setting-up-prow-for-a-new-repo-reviewers-assignment-and-auto-merge) in Github UI other than tide, this is theoretically not necessary since Prow jobs already have a field of [`optional: true|false`](https://github.com/knative/test-infra/blob/908ad6cdb8e4aab96cdf0d94f766db7caf4fe129/config/prow/jobs/config.yaml#L418), and this is a duplicate of effort with exactly the same effect

We should be able to just configure tide as required check in Github UI, and before doing that, there  are several things need to be done first:

1. Github UI only labels a job as `required` when it's checked in Github UI, this will cause confusion to developers trying to figure out which job is required vs. optional. This could be possibly solved by  Prow adding `[required]` or `[optional]` in job status report
1. Currently Prow jobs are configured correctly for master branches, but not necessarily true with old release branches. i.e. latest added Prow job [`pull-knative-serving-autotls-tests`](https://github.com/knative/test-infra/blob/908ad6cdb8e4aab96cdf0d94f766db7caf4fe129/config/prow/jobs/config.yaml#L378) was configured to be required for all branches, which doesn't exist in old release branches
